volatility3/framework/plugins/windows/cmdscan.py

```diff
@@ -0,0 +1,380 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+# This module attempts to locate windows console histories.
+
+import logging
+import struct
+from typing import Tuple, Generator, Set, Dict, Any, List, Optional
+
+from volatility3.framework import interfaces
+from volatility3.framework import renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.layers import scanners
+from volatility3.framework.objects import utility
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import pslist, consoles
+
+
+vollog = logging.getLogger(__name__)
+
+
+class CmdScan(interfaces.plugins.PluginInterface):
+    """Looks for Windows Command History lists"""
+
+    _required_framework_version = (2, 4, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.PluginRequirement(
+                name="consoles", plugin=consoles.Consoles, version=(1, 0, 0)
+            ),
+            requirements.BooleanRequirement(
+                name="no_registry",
+                description="Don't search the registry for possible values of CommandHistorySize",
+                optional=True,
+                default=False,
+            ),
+            requirements.ListRequirement(
```

Comment:
Also, why is the default `[50]`?  Will that return responses in most cases?  Should we not have a default value that returns *all* `CommandHistory` sizes?

Reply:
I was trying to make it flexible. In probably 99% of the cases it will always be 50. I've added the ability, which wasn't in vol2, for the code to walk the Registry for other values. That value is stored in a per user hive in the Registry, so it could theoretically be anything, though in practice it's unlikely people change it.  On the off chance it was changed, and the registry scan found a value of, say, 100, then having to to 50 < x < 100 or something would cause the scanning for find a whole range of potential values, rather than the 2 we want to look for.